### ISSUE TITLE
Self provided normalization function is not used.

### ISSUE THREAD

**Issue by cosmozhang** [RaRe-Technologies/gensim#3244]
https://github.com/RaRe-Technologies/gensim/blob/5bec27767ad40712e8912d53a896cb2282c33880/gensim/models/tfidfmodel.py#L525

`self.normalize = matutils.unitvec` does not allow users to use self-defined normalization function.

**Comment by piskvorky** [comment]
Correct! If you found a bug, please fill in the issue template fully. Otherwise I'll close this.

**Comment by gojomo** [comment]
@cosmozhang Yes - PRs for consideration should generally be against the `develop` branch. Unless the problem & fix is trivial/self-evident, it may also help to include inside the PR a test case that fails before, and succeeds after, the fix is applied.
